internal/signer/darwin/keychain/keychain.go

```diff
@@ -141,15 +150,19 @@ type Key struct {
 	privateKeyRef C.SecKeyRef
 	certs         []*x509.Certificate
 	once          sync.Once
+	publicKeyRef  C.SecKeyRef
+	hashFunc      crypto.Hash
 }
 
 // newKey makes a new Key wrapper around the key reference,
 // takes ownership of the reference, and sets up a finalizer to handle releasing
 // the reference.
-func newKey(privateKeyRef C.SecKeyRef, certs []*x509.Certificate) (*Key, error) {
+func newKey(privateKeyRef C.SecKeyRef, certs []*x509.Certificate, publicKeyRef C.SecKeyRef) (*Key, error) {
```

Comment:
This func needs to also be updated to retain the public key.

Also update the Close command to free the public key ref